Mathlib/Analysis/InnerProductSpace/OrthoDecomp.lean

```diff
@@ -0,0 +1,249 @@
+import Mathlib.Analysis.InnerProductSpace.Projection
+import Mathlib.Analysis.InnerProductSpace.ProdL2
+import Mathlib.MeasureTheory.Measure.Haar.InnerProductSpace
+import Mathlib.MeasureTheory.Integral.Bochner
+
+open IsROrC Real Filter
+namespace InnerProductSpace
+
+variable {ι₁ ι₂ 𝕜 E F A : Type*}
+
+noncomputable section prodBasis
+
+variable [IsROrC 𝕜] [Fintype ι₁] [Fintype ι₂]
+variable [NormedAddCommGroup E] [InnerProductSpace 𝕜 E]
+variable [NormedAddCommGroup F] [InnerProductSpace 𝕜 F]
+
+def _root_.OrthonormalBasis.prod (v : OrthonormalBasis ι₁ 𝕜 E) (w : OrthonormalBasis ι₂ 𝕜 F) :
+    OrthonormalBasis (ι₁ ⊕ ι₂) 𝕜 (WithLp 2 (E × F)) :=
+  ((v.toBasis.prod w.toBasis).map (WithLp.linearEquiv 2 𝕜 (E × F)).symm).toOrthonormalBasis
+  (by
+    constructor
+    · simp [Sum.forall, norm_eq_sqrt_inner (𝕜 := 𝕜), Real.sqrt_eq_one]
+      simp [← sqrt_eq_one, ← norm_eq_sqrt_inner (𝕜 := 𝕜), v.orthonormal.1, w.orthonormal.1]
+    · unfold Pairwise
+      simp [Sum.forall]
+      exact ⟨v.orthonormal.2, w.orthonormal.2⟩)
+
+@[simp] theorem _root_.OrthonormalBasis.prod_apply (v : OrthonormalBasis ι₁ 𝕜 E)
+    (w : OrthonormalBasis ι₂ 𝕜 F) :
+    ∀ i : ι₁ ⊕ ι₂, v.prod w i = Sum.elim ((LinearMap.inl 𝕜 E F) ∘ v) ((LinearMap.inr 𝕜 E F) ∘ w) i := by
+  rw [Sum.forall]
+  unfold OrthonormalBasis.prod
+  constructor
+  · intro
+    simp
```

Comment:
Does `simpa` work in place of simp then rfl? (Iirc it does but I could be wrong) That might be less brittle if so.